range_queries/prefix_sum_array.cpp

```diff
@@ -0,0 +1,69 @@
+/**
+ * @file
+ * @brief
+ * Implementation of [Prefix Sum
+ * Array](https://en.wikipedia.org/wiki/Prefix_sum) data structure.
+ *
+ * @details
+ * Prefix Sum Array is a data structure, that allows answering sum in some range
+ * queries. It can answer most sum range queries in O(1), with a build time
+ * complexity of O(N). But it hasn't an update querie.
+ *
+ * * Running Time Complexity \n
+ * * Build : O(N) \n
+ * * Range Query : O(1) \n
+ */
+
+#include <cassert>   // for assert function
+#include <iostream>  // for IO operations
+#include <vector>    // for using the STL vector data structure
+
+/**
+ * @namespace range_queries
+ * @brief Range Queries algorithms
+ */
+namespace range_queries {
+/**
+ * @namespace prefix_sum_array
+ * @brief Range sum queries using prefix-sum-array
+ */
+namespace prefix_sum_array {
+
+std::vector<int> PSA(1, 0);
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂